kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/codec/KafkaRequestDecoder.java

```diff
@@ -62,6 +62,12 @@ protected Frame decodeHeaderAndBody(ChannelHandlerContext ctx, ByteBuf in, final
         if (log().isTraceEnabled()) { // avoid boxing
             log().trace("{}: apiVersion: {}", ctx, apiVersion);
         }
+        short lowestSupportedVersion = apiKey.messageType.lowestSupportedVersion();
```

Comment:
there was a promising looking method:

```
public boolean isVersionDeprecated(short apiVersion) {
        return apiVersion >= messageType.lowestDeprecatedVersion() && apiVersion <= messageType.highestDeprecatedVersion();
    }
```

on ApiKeys, but it didn't behave as expected because the `highestDeprecatedVersion` is set to `-1`. The method is only used to control log levels by the look, so I wonder if this is a problem upstream nobody has noticed.

Reply:
Yes, I think `isVersionSupported` is doing the similar thing here. So maybe we also want to throw `UnsupportedVersionException` when `apiVersion > highestSupportedVersion`?

Reply:
I see. Let me have another look.

Reply:
Thanks @showuon